src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineFlexibleStaging.java

```diff
@@ -73,16 +75,17 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
     }
   }
 
-  private static String findRuntime(StageFlexibleConfiguration config) {
+  @VisibleForTesting
+  static String findRuntime(StageFlexibleConfiguration config) {
     // verification for app.yaml that contains runtime:java
     Path appYaml = config.getAppEngineDirectory().toPath().resolve(APP_YAML);
     String runtime = null;
     try {
       if (Files.isRegularFile(appYaml)) {
         runtime = new AppYaml(appYaml).getRuntime();
       }
-    } catch (IOException e) {
-      log.warning("Unable to determine runtime: error parsing app.yaml : " + e.getMessage());
+    } catch (IOException | ScannerException | ParserException ex) {
```

Comment:
You mean let `findRuntime()` throw these exceptions (for that matter, I think we don't really need a new `InvalidAppYamlException`) and let `copyAppEngineContext()` (which is the caller) to swallow it, right? Or, do you propagate such an exception to clients of appengine-plugins-core, meaning staging failure?